layers/place/city.sql

```diff
@@ -56,22 +56,22 @@ FROM (
                          row_number() OVER (
                              PARTITION BY LabelGrid(geometry, 128 * pixel_width)
                              ORDER BY "rank" ASC NULLS LAST,
-                                 place ASC NULLS LAST,
+                                 place ASC NULLS LAST, -- place here is type of place, one of 'city', 'town', 'village', 'hamlet', 'suburb', 'quarter', 'neighbourhood', 'isolated_dwelling'
                                  population DESC NULLS LAST,
                                  length(name) ASC
                              )::int AS gridrank
                   FROM osm_city_point
                   WHERE geometry && bbox
-                    AND ((zoom_level = 7 AND place <= 'town'::city_place
+                    AND ((zoom_level BETWEEN 4 AND 7 AND place <= 'town'::city_place
```

Comment:
Does this just expand the intermediate results for the query, which are then filtered in the WHERE clause on line 71?

Reply:
Yes, without this anything without an OSM "rank" is excluded below zoom 7 (line 34)